extensions/axon/deployment/src/main/java/io/quarkus/axon/deployment/AxonProcessor.java

```diff
@@ -0,0 +1,172 @@
+package io.quarkus.axon.deployment;
+
+import static io.quarkus.deployment.annotations.ExecutionTime.RUNTIME_INIT;
+import static io.quarkus.deployment.annotations.ExecutionTime.STATIC_INIT;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+
+import org.axonframework.commandhandling.CommandHandler;
+import org.axonframework.eventhandling.EventHandler;
+import org.axonframework.modelling.command.AggregateRoot;
+import org.axonframework.modelling.saga.StartSaga;
+import org.axonframework.queryhandling.QueryHandler;
+import org.jboss.jandex.*;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.BeanArchiveIndexBuildItem;
+import io.quarkus.arc.deployment.BeanContainerBuildItem;
+import io.quarkus.axon.runtime.AxonClientProducer;
+import io.quarkus.axon.runtime.AxonRuntimeConfig;
+import io.quarkus.axon.runtime.AxonRuntimeTemplate;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+
+public class AxonProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(AxonProcessor.class);
+
+    private static DotName AGGREGATE_ANNOTATION = DotName.createSimple(AggregateRoot.class.getName());
+    private static DotName SAGA_ANNOTATION = DotName.createSimple(StartSaga.class.getName());
+    private static DotName COMMAND_HANDLER_ANNOTATION = DotName.createSimple(CommandHandler.class.getName());
+    private static DotName EVENT_HANDLER_ANNOTATION = DotName.createSimple(EventHandler.class.getName());
+    private static DotName QUERY_HANDLER_ANNOTATION = DotName.createSimple(QueryHandler.class.getName());
+
+    @BuildStep
+    @Record(STATIC_INIT)
```

Comment:
The `@Record` annotation should only be used when one of the arguments is a "Recorder", meaning a class that is annotated with `@Recorder` (or the deprecated `@Template` like you have in `AxonRuntimeTemplate`). 
So please go over the `BuildStep`s you have included and remove all unnecessary `@Record` annotations.